### PR TITLE
Feature/TK 5: Develop Front-end Integration for Online or Offline Screen Flow

### PR DESCRIPTION
**Pull request by illuminatyy**
# What was done? ✅
> Describe in a few words what have you done? 
> - Step 1: Created an `@State var isOnline` to passthrough the online state(boolean) in all pages/steps of creation
> - Step 2: Added the new variable on Navigation links
> - Step 3: Added on call to server the `isOnline` state on `QCAmountWinRulesView.swift` button "Criar desafio"
> - Step 4: Changed the `var invitationCode` for optional in `QuickChallenge.swift` model
> - Step 5: Fixed `enum ErrorState var description` to return a `LocalizedStringKey` and modified localizables files to change the alert error titles in English

# Description 📝
  
# Evidence 🕵️‍♀️
## **Screenshots/Videos/Figma** 📱
- Online Flow:
<video src="https://user-images.githubusercontent.com/51174686/198120553-080497e8-1d38-48c4-97d4-6599b2223b02.mp4" width=414 controls>

Captura de Tela 2022-10-26 às 00 26 19

- Offline Flow:
<video src="https://user-images.githubusercontent.com/51174686/198120662-258ea4b7-51c0-4be5-8575-8eec1a1bd176.mp4" width=414 controls>

Captura de Tela 2022-10-26 às 00 27 54

### CODE REVIEW

File: Fiero/Views/Challenges/Quick/QCNamingView.swift [hunk 1]
```diff
@@ -14,6 +14,7 @@ struct QCNamingView: View {
     @State private var isNavActiveForAmount: Bool = false
     @State var challengeName: String = ""
     @State var isPresentingAlert: Bool = false
+    @State var isOnline: Bool
```
Comment (MarinaFX) [inline Fiero/Views/Challenges/Quick/QCNamingView.swift:17]: Como nao vamos mais mudar o estado desse booleano (nao vamos alterar na tela de nome do desafio se ele eh online ou offline), ele pode ser uma var normal. 

Ele sendo declarado como `@State` aqui, acaba criando duas fontes da verdade (o state da tela anterior e o de agora), e mais pra frente pode nos dar problema de referencias

File: Fiero/Views/Challenges/Quick/Amount/QCAmountWinRulesView.swift [hunk 2]
```diff
@@ -34,6 +34,7 @@ struct QCAmountWinRulesView: View {
     @State private var subscriptions: Set<AnyCancellable> = []
     @State private var errorState: ErrorState?
     
+    @State var isOnline: Bool
```
Comment (MarinaFX) [inline Fiero/Views/Challenges/Quick/Amount/QCAmountWinRulesView.swift:37]: aqui tbm

**Review by MarinaFX — CHANGES_REQUESTED** [review]

File: Fiero/Views/Challenges/Quick/QCNamingView.swift (hunk 1 above)
Comment (illuminatyy) [reply Fiero/Views/Challenges/Quick/QCNamingView.swift]: feito!

File: Fiero/Views/Challenges/Quick/Amount/QCAmountWinRulesView.swift (hunk 2 above)
Comment (illuminatyy) [reply Fiero/Views/Challenges/Quick/Amount/QCAmountWinRulesView.swift]: feito tbm

**Review by MarinaFX — APPROVED** [review]

**Pull request merged**